amespahdbpythonsuite/spectrum.py

```diff
@@ -362,3 +363,94 @@ def resample(self, grid: np.ndarray) -> None:
             self.data[uid] = s.flux.value
 
         self.grid = grid
+
+    def mcfit(self, obs, nsamples, **keywords) -> Mcfitted:
+        """
+        Monte Carlo sampling and fitting to the input spectrum.
+
+        Properties
+        ----------
+        obs : observations object
+
+        nsamples : Number of samples.
+            int
+
+        """
+        from amespahdbpythonsuite.mcfitted import Mcfitted
+
+        # Initialize lists and dictionaries.
+        fits = []
+        classes = []
+        weights = []
+        results = {'solo': [],
+                   'duo': [],
+                   'trio': [],
+                   'quartet': [],
+                   'quintet': [],
+                   'anion': [],
+                   'neutral': [],
+                   'cation': [],
+                   'small': [],
+                   'large': [],
+                   'nitrogen': [],
+                   'pure': [],
+                   'nc': [],
+                   'error': []
+                   }
+
+        keys = list(results.keys())
+
+        if 'predict' in keywords:
+            predicted = []
+        else:
+            predicted = False
+
+        # Start the MC sampling and fitting.
+        for i in range(nsamples):
+            print(f'MC sampling {i+1}/{nsamples}')
+
+            # Calculate new flux based on random uniform distribution sampling.
+            flux = np.random.normal(obs.spectrum.flux.value, obs.spectrum.uncertainty.array)\
+                * obs.spectrum.flux.unit
+
+            # Fit the spectrum.
+            fit = self.fit(flux, obs.spectrum.uncertainty)
+
+            # Obtain the fit and weights.
+            fits.append(fit.getfit())
+            weights.append(fit.getweights())
+
+            # Obtain fit breakdown.
+            bd = fit.getbreakdown()
+            for key in keys[:-1]:
+                results[key].append(bd[key])
+
+            # Obtain fit uncertainty.
+            error = fit.geterror()['err']
+            results['error'].append(error)
+
+            # Obtain classes.
+            classes.append(fit.getclasses())
+
+            if 'predict' in keywords:
```

Comment:
A method could be added to Fitted that does the predicting. In any case, here this feature needs to be moved to MCFitted.